lib/postprocessing/image.js

```diff
@@ -639,22 +637,55 @@ function convertSvg(img, instructions) {
     // Only using width because ImageMagick will use the smallest value whether it be width or height
     const width = instructions.width || SVG_DEFAULT_WIDTH;
 
-    img = handleTransparency(img, instructions.fmt);
+    img = handleTransparency(img, instructions);
 
     // some svgs have no size (only percentage width/height), so we scale them to our target size
     img.in("-size", `${width}`); // 2020-11-10 img.rawSize() will not be applied at the correct time and the SVG will be upscaled resulting in a fuzzy final rendition
     return img;
 }
 
-// gm("img.png").set(attribute, value)
-function handleTransparency(img, fmt) {
-    if (FORMAT_SUPPORTS_TRANSPARENCY[fmt]) {
-        img.background(DEFAULT_TRANSPARENCCY_BACKGROUND); // Keep transparency
+function parseBackgroundColor(instructions){
+    let inputBackgroundColor = instructions.pdfbgcolor;
+    if(!inputBackgroundColor) return null;
+
+    if(typeof inputBackgroundColor === 'string') {
+        if(inputBackgroundColor.match(/^#([A-Fa-f0-9]{6}|[A-Fa-f0-9]{3})$/)){
+            // HEX color
+            return inputBackgroundColor;
+        } else if(inputBackgroundColor.match(/^\d+$/)) {
+            // Possibly the numerical value to apply to the R, G and B channels
+            inputBackgroundColor = Number.parseInt(inputBackgroundColor, 10);
+        } else {
+            throw new GenericError(`Invalid background color set in instructions (value set was: ${instructions.pdfbgcolor})`);
+        }
+    }
+
+    // If we are here, we have numerical values for the color channels
+    // Verify the numerical value to apply to the R, G and B channels
+    if(Number.isInteger(inputBackgroundColor)
+      && inputBackgroundColor >= 0
+      && inputBackgroundColor <= 255){
+        return `rgb(${inputBackgroundColor},${inputBackgroundColor},${inputBackgroundColor})`;
     } else {
-        img.background(DEFAULT_BACKGROUND_COLOR); // Change tranparency background to default background color 
-        img.flatten();  // forces the image on its background to "commit" the background
+        throw new GenericError(`Invalid background color values set for RGB channels in instructions: ${instructions.pdfbgcolor}`);
     }
+}
 
+function handleTransparency(img, instructions) {
+    if(instructions.pdfbgcolor) {
```

Comment:
if we expect the supported format to be `png` then the condition should specify it. 
if(instructions.pdfbgcolor && instructions.fmt === 'png') {

Reply:
at that point we'd support anything, since the whole imagemagick command will do the conversion. It's the input we expect to be PNG.